In [191]:
import pandas as pd
import numpy as np

In [261]:
rest = pd.read_csv('TA_restaurants_curated.csv',encoding='utf8', index_col=0)
rest.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 125527 entries, 0 to 1666
Data columns (total 10 columns):
Name                 125527 non-null object
City                 125527 non-null object
Cuisine Style        94176 non-null object
Ranking              115876 non-null float64
Rating               115897 non-null float64
Price Range          77672 non-null object
Number of Reviews    108183 non-null float64
Reviews              115911 non-null object
URL_TA               125527 non-null object
ID_TA                125527 non-null object
dtypes: float64(3), object(7)
memory usage: 10.5+ MB


In [262]:
print(len(rest))
rest.head()

125527


Name       City  \
0  Martine of Martine's Table  Amsterdam   
1         De Silveren Spiegel  Amsterdam   
2                     La Rive  Amsterdam   
3                    Vinkeles  Amsterdam   
4   Librije's Zusje Amsterdam  Amsterdam   

                                       Cuisine Style  Ranking  Rating  \
0                    ['French', 'Dutch', 'European']     1.00    5.00   
1  ['Dutch', 'European', 'Vegetarian Friendly', '...     2.00    4.50   
2  ['Mediterranean', 'French', 'International', '...     3.00    4.50   
3  ['French', 'European', 'International', 'Conte...     4.00    5.00   
4  ['Dutch', 'European', 'International', 'Vegeta...     5.00    4.50   

  Price Range  Number of Reviews  \
0    $$ - $$$             136.00   
1        $$$$             812.00   
2        $$$$             567.00   
3        $$$$             564.00   
4        $$$$             316.00   

                                             Reviews  \
0  [['Just like home', 'A Warm Welcome to Wintry ...   
1  [['Great food and staff', 'just perfect'], ['0...   
2  [['Satisfaction', 'Delicious old school restau...   
3  [['True five star dinner', 'A superb evening o...   
4  [['Best meal.... EVER', 'super food experience...   

                                              URL_TA      ID_TA  
0  /Restaurant_Review-g188590-d11752080-Reviews-M...  d11752080  
1  /Restaurant_Review-g188590-d693419-Reviews-De_...    d693419  
2  /Restaurant_Review-g188590-d696959-Reviews-La_...    d696959  
3  /Restaurant_Review-g188590-d1239229-Reviews-Vi...   d1239229  
4  /Restaurant_Review-g188590-d6864170-Reviews-Li...   d6864170

In [263]:
pd.options.display.float_format = '{:.2f}'.format
rest.describe()

Ranking    Rating  Number of Reviews
count 115876.00 115897.00          108183.00
mean    3657.46      3.99             125.18
std     3706.26      0.68             310.83
min        1.00     -1.00               2.00
25%      965.00      3.50               9.00
50%     2256.00      4.00              32.00
75%     5237.00      4.50             114.00
max    16444.00      5.00           16478.00

In [264]:
rest['Number of Reviews'] = rest['Number of Reviews'].fillna(0)
rest['Number of Reviews'] = rest['Number of Reviews'].round(0).astype('int')
rest = rest.drop_duplicates('ID_TA', keep='first')
rest.Rating.replace(-1, 0, inplace=True)

In [265]:
rest.info()
rest.dtypes

<class 'pandas.core.frame.DataFrame'>
Int64Index: 125326 entries, 0 to 1666
Data columns (total 10 columns):
Name                 125326 non-null object
City                 125326 non-null object
Cuisine Style        94047 non-null object
Ranking              115710 non-null category
Rating               115734 non-null float64
Price Range          77574 non-null object
Number of Reviews    125326 non-null int64
Reviews              115745 non-null object
URL_TA               125326 non-null object
ID_TA                125326 non-null object
dtypes: category(1), float64(1), int64(1), object(7)
memory usage: 10.6+ MB


Name                   object
City                   object
Cuisine Style          object
Ranking              category
Rating                float64
Price Range            object
Number of Reviews       int64
Reviews                object
URL_TA                 object
ID_TA                  object
dtype: object

In [266]:
rest.rename(columns={'Name': 'name',
            'City': 'city',
            'Ranking': 'ranking',
            'Rating': 'rating',
            'Reviews': 'reviews',
            'Cuisine Style':'cuisine_style',
            'Price Range':'price_range',
            'Number of Reviews':'reviews_number'}, inplace=True)

In [267]:
rest.head()

name       city  \
0  Martine of Martine's Table  Amsterdam   
1         De Silveren Spiegel  Amsterdam   
2                     La Rive  Amsterdam   
3                    Vinkeles  Amsterdam   
4   Librije's Zusje Amsterdam  Amsterdam   

                                       cuisine_style ranking  rating  \
0                    ['French', 'Dutch', 'European']    1.00    5.00   
1  ['Dutch', 'European', 'Vegetarian Friendly', '...    2.00    4.50   
2  ['Mediterranean', 'French', 'International', '...    3.00    4.50   
3  ['French', 'European', 'International', 'Conte...    4.00    5.00   
4  ['Dutch', 'European', 'International', 'Vegeta...    5.00    4.50   

  price_range  reviews_number  \
0    $$ - $$$             136   
1        $$$$             812   
2        $$$$             567   
3        $$$$             564   
4        $$$$             316   

                                             reviews  \
0  [['Just like home', 'A Warm Welcome to Wintry ...   
1  [['Great food and staff', 'just perfect'], ['0...   
2  [['Satisfaction', 'Delicious old school restau...   
3  [['True five star dinner', 'A superb evening o...   
4  [['Best meal.... EVER', 'super food experience...   

                                              URL_TA      ID_TA  
0  /Restaurant_Review-g188590-d11752080-Reviews-M...  d11752080  
1  /Restaurant_Review-g188590-d693419-Reviews-De_...    d693419  
2  /Restaurant_Review-g188590-d696959-Reviews-La_...    d696959  
3  /Restaurant_Review-g188590-d1239229-Reviews-Vi...   d1239229  
4  /Restaurant_Review-g188590-d6864170-Reviews-Li...   d6864170

In [268]:
print("Número de Ciudades en el df: ", rest.city.nunique())
print("Número de Restaurants en el df: ", rest.ID_TA.nunique())

Número de Ciudades en el df:  31
Número de Restaurants en el df:  125326


In [269]:
df = rest.copy()

### TOP 10 Ciudades con más Restaurants

In [270]:
top_cities = (df.groupby('city')['name'].count().to_frame()).sort_values(by='name',ascending = False).head(n=10)
top_cities2 = list(top_cities.index)
top_cities2

['London',
 'Paris',
 'Madrid',
 'Barcelona',
 'Berlin',
 'Milan',
 'Rome',
 'Prague',
 'Lisbon',
 'Vienna']

In [271]:
df_filtered = df[df['city'].isin(top_cities2)]
df_filtered

name       city  \
0                        Uma  Barcelona   
1                      Viana  Barcelona   
2                     Blavis  Barcelona   
3              My Restaurant  Barcelona   
4            Bodega Biarritz  Barcelona   
...                      ...        ...   
3719  restaurant you, Vienne     Vienna   
3720       Kartonage Kitchen     Vienna   
3721                    Fino     Vienna   
3722       Cafe Zeitgenossin     Vienna   
3723               El Gaucho     Vienna   

                                          cuisine_style ranking  rating  \
0     ['International', 'Mediterranean', 'Fusion', '...    1.00    5.00   
1     ['International', 'Mediterranean', 'Spanish', ...    2.00    5.00   
2     ['Mediterranean', 'European', 'Spanish', 'Vege...    3.00    5.00   
3     ['Mediterranean', 'European', 'Spanish', 'Vege...    4.00    5.00   
4     ['Mediterranean', 'Spanish', 'Wine Bar', 'Euro...    5.00    5.00   
...                                                 ...     ...     ...   
3719                 ['French', 'Austrian', 'European']     nan     nan   
3720  ['Austrian', 'International', 'European', 'Fus...     nan     nan   
3721                 ['German', 'Austrian', 'European']     nan     nan   
3722                  ['Cafe', 'Contemporary', 'Diner']     nan     nan   
3723                                                NaN     nan     nan   

     price_range  reviews_number  \
0           $$$$             792   
1       $$ - $$$            2707   
2       $$ - $$$             643   
3       $$ - $$$             159   
4       $$ - $$$            1078   
...          ...             ...   
3719         NaN               0   
3720           $               0   
3721         NaN               0   
3722           $               0   
3723         NaN               0   

                                                reviews  \
0     [['Perfect atmosphere and location', 'Perfecti...   
1     [['Wow! Best ever!', 'Small menu-- GET A RESER...   
2     [['The whole package', 'Incredible Service and...   
3     [['Fantastic evening!!', 'Perfect meal'], ['01...   
4     [['Deserves its reputation!', 'THESE ARE THE T...   
...                                                 ...   
3719                                                NaN   
3720                                                NaN   
3721                                                NaN   
3722                                                NaN   
3723                                                NaN   

                                                 URL_TA      ID_TA  
0     /Restaurant_Review-g187497-d8003030-Reviews-Um...   d8003030  
1     /Restaurant_Review-g187497-d8531409-Reviews-Vi...   d8531409  
2     /Restaurant_Review-g187497-d3353672-Reviews-Bl...   d3353672  
3     /Restaurant_Review-g187497-d12643500-Reviews-M...  d12643500  
4     /Restaurant_Review-g187497-d11906544-Reviews-B...  d11906544  
...                                                 ...        ...  
3719  /Restaurant_Review-g190454-d13331692-Reviews-R...  d13331692  
3720  /Restaurant_Review-g190454-d13337747-Reviews-K...  d13337747  
3721  /Restaurant_Review-g190454-d13341715-Reviews-F...  d13341715  
3722  /Restaurant_Review-g190454-d13343747-Reviews-C...  d13343747  
3723  /Restaurant_Review-g190454-d13359780-Reviews-E...  d13359780  

[83149 rows x 10 columns]

In [272]:
# Función MAP para evaluar el rango de precios en categorías.

df_filtered['price_range'] = df_filtered['price_range'].fillna('NA')
price_ranges = {'$': 'Barato', '$$ - $$$': 'Medio', '$$$$': 'Costoso', 'NA': 'SinInformacion'}
df_filtered['price_range'] = df_filtered.price_range.map(price_ranges)
df_filtered

/Users/marianalara/opt/anaconda3/lib/python3.7/site-packages/ipykernel_launcher.py:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  This is separate from the ipykernel package so we can avoid doing imports until
/Users/marianalara/opt/anaconda3/lib/python3.7/site-packages/ipykernel_launcher.py:5: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  """


name       city  \
0                        Uma  Barcelona   
1                      Viana  Barcelona   
2                     Blavis  Barcelona   
3              My Restaurant  Barcelona   
4            Bodega Biarritz  Barcelona   
...                      ...        ...   
3719  restaurant you, Vienne     Vienna   
3720       Kartonage Kitchen     Vienna   
3721                    Fino     Vienna   
3722       Cafe Zeitgenossin     Vienna   
3723               El Gaucho     Vienna   

                                          cuisine_style ranking  rating  \
0     ['International', 'Mediterranean', 'Fusion', '...    1.00    5.00   
1     ['International', 'Mediterranean', 'Spanish', ...    2.00    5.00   
2     ['Mediterranean', 'European', 'Spanish', 'Vege...    3.00    5.00   
3     ['Mediterranean', 'European', 'Spanish', 'Vege...    4.00    5.00   
4     ['Mediterranean', 'Spanish', 'Wine Bar', 'Euro...    5.00    5.00   
...                                                 ...     ...     ...   
3719                 ['French', 'Austrian', 'European']     nan     nan   
3720  ['Austrian', 'International', 'European', 'Fus...     nan     nan   
3721                 ['German', 'Austrian', 'European']     nan     nan   
3722                  ['Cafe', 'Contemporary', 'Diner']     nan     nan   
3723                                                NaN     nan     nan   

         price_range  reviews_number  \
0            Costoso             792   
1              Medio            2707   
2              Medio             643   
3              Medio             159   
4              Medio            1078   
...              ...             ...   
3719  SinInformacion               0   
3720          Barato               0   
3721  SinInformacion               0   
3722          Barato               0   
3723  SinInformacion               0   

                                                reviews  \
0     [['Perfect atmosphere and location', 'Perfecti...   
1     [['Wow! Best ever!', 'Small menu-- GET A RESER...   
2     [['The whole package', 'Incredible Service and...   
3     [['Fantastic evening!!', 'Perfect meal'], ['01...   
4     [['Deserves its reputation!', 'THESE ARE THE T...   
...                                                 ...   
3719                                                NaN   
3720                                                NaN   
3721                                                NaN   
3722                                                NaN   
3723                                                NaN   

                                                 URL_TA      ID_TA  
0     /Restaurant_Review-g187497-d8003030-Reviews-Um...   d8003030  
1     /Restaurant_Review-g187497-d8531409-Reviews-Vi...   d8531409  
2     /Restaurant_Review-g187497-d3353672-Reviews-Bl...   d3353672  
3     /Restaurant_Review-g187497-d12643500-Reviews-M...  d12643500  
4     /Restaurant_Review-g187497-d11906544-Reviews-B...  d11906544  
...                                                 ...        ...  
3719  /Restaurant_Review-g190454-d13331692-Reviews-R...  d13331692  
3720  /Restaurant_Review-g190454-d13337747-Reviews-K...  d13337747  
3721  /Restaurant_Review-g190454-d13341715-Reviews-F...  d13341715  
3722  /Restaurant_Review-g190454-d13343747-Reviews-C...  d13343747  
3723  /Restaurant_Review-g190454-d13359780-Reviews-E...  d13359780  

[83149 rows x 10 columns]

In [273]:
rango_precio = df_filtered.groupby(['city','price_range'])[['name']].count()
rating_mean = df_filtered.groupby(['city','price_range'])[['rating']].mean()

In [274]:
rango_rating = rango_precio.join(rating_mean, rsuffix = ('_precio'))
rango_rating.rename(columns = {'name' : 'num_rest', 'rating':'rating_promedio'}, inplace = True)
rango_rating.sort_index()

num_rest  rating_promedio
city      price_range                              
Barcelona Barato              1464             4.04
          Costoso              263             4.19
          Medio               3680             4.00
          SinInformacion      3018             3.86
Berlin    Barato              1153             4.27
          Costoso              134             4.31
          Medio               2366             4.08
          SinInformacion      3425             4.10
Lisbon    Barato               766             4.14
          Costoso              109             4.14
          Medio               1638             4.04
          SinInformacion      1473             4.02
London    Barato              2985             4.03
          Costoso              648             4.14
          Medio               8528             3.95
          SinInformacion      6038             3.86
Madrid    Barato              1044             3.84
          Costoso              300             4.09
          Medio               3535             3.89
          SinInformacion      4498             3.67
Milan     Barato              1241             3.89
          Costoso              243             4.11
          Medio               2337             3.84
          SinInformacion      2859             3.71
Paris     Barato              1726             4.03
          Costoso              679             4.17
          Medio               7175             3.95
          SinInformacion      5294             3.89
Prague    Barato               871             4.12
          Costoso              101             4.35
          Medio               1702             4.02
          SinInformacion      2185             3.93
Rome      Barato              1472             4.31
          Costoso              225             4.26
          Medio               3004             4.09
          SinInformacion      1248             4.47
Vienna    Barato               392             4.16
          Costoso              108             4.20
          Medio               1627             4.04
          SinInformacion      1595             4.07

### Evaluación de los rest Costosos:
- **Paris** tiene la mayor cantidad de restaurants costosos.
- **Praga** tiene la menor catidad.
- **Praga** tiene el mejor rating promedio con 4.35
- **Madrid** tiene el promedio de rating más bajo de este grupo con 4.09

In [275]:
costosos = rango_rating.xs('Costoso',level = 'price_range').sort_values(by='num_rest',ascending= False)
costosos.style.highlight_max(axis=0,color='lightgreen').highlight_min(axis=0,color='pink')

### Evaluación de los rest Medios:
- **London** tiene la mayor cantidad de restaurants en el rango medio.
- **Vienna** tiene la menor catidad.
- **Roma** tiene el mejor rating promedio con 4.09
- **Milan** tiene el promedio de rating más bajo de este grupo con 3.83

In [276]:
medio = rango_rating.xs('Medio',level = 'price_range').sort_values(by='num_rest',ascending= False)
medio.style.highlight_max(axis=0,color='lightgreen').highlight_min(axis=0,color='pink')

### Evaluación de los rest Baratos:
- **Londres** tiene la mayor cantidad de restaurants baratos.
- **Vienna** tiene la menor catidad.
- **Roma** tiene el mejor rating promedio con 4.31
- **Madrid** tiene el promedio de rating más bajo de este grupo con 3.83

In [277]:
baratos = rango_rating.xs('Barato',level = 'price_range').sort_values(by='num_rest',ascending= False)
baratos.style.highlight_max(axis=0,color='lightgreen').highlight_min(axis=0,color='pink')

### Evaluación de los rest "Sin Información":
- **Londres** tiene la mayor cantidad de restaurants sin información.
- **Roma** tiene la menor catidad.
- **Roma** tiene el mejor rating promedio con 4.45
- **Madrid** tiene el promedio de rating más bajo de este grupo con 3.67

In [278]:
sininfor = rango_rating.xs('SinInformacion',level = 'price_range').sort_values(by='num_rest',ascending= False)
sininfor.style.highlight_max(axis=0,color='lightgreen').highlight_min(axis=0,color='pink')


In [279]:
tmp = (df_filtered.groupby(['city']).agg({'rating': ['mean','max','min']}))
tmp.sort_values(by=('rating', 'mean'),ascending=False)

rating          
            mean  max  min
city                      
Rome        4.23 5.00 3.00
Berlin      4.13 5.00 0.00
Vienna      4.07 5.00 0.00
Lisbon      4.05 5.00 1.00
Prague      4.01 5.00 0.00
Barcelona   3.97 5.00 0.00
Paris       3.95 5.00 0.00
London      3.94 5.00 0.00
Milan       3.81 5.00 0.00
Madrid      3.79 5.00 0.00

Del DF anterior, podemos concluir que Roma tiene el mejor promedio (*no es sorpresa para mi*), Berlin se posiciona en el segund luar y Vienna en el tercero. 
Madrid se encuentra al final de esta tabla con un 3.79 en promedio.
La desviación estandar en la evaluación promedio de estos grupos es de 0.13 puntos.

In [290]:
cocinas = df_filtered.loc[:, ['city', 'name', 'cuisine_style', 'rating', 'reviews_number', 'reviews','price_range']]
#Elimino los corchetes
cocinas.cuisine_style = cocinas.cuisine_style.str.replace('[', '')
cocinas.cuisine_style = cocinas.cuisine_style.str.replace(']', '')
cocinas.columns = ['city', 'place', 'cuisine_style', 'rating', 'reviews_number', 'reviews','price_range']
#Filtro las cocinas por aquellas que tengan su rating > a 5 y numero de reviews superior a la media
cocinas = cocinas[(cocinas['reviews_number'] >= cocinas.reviews_number.mean())&(cocinas['rating'] >= 5)]
cocinas

city                      place  \
0     Barcelona                        Uma   
1     Barcelona                      Viana   
2     Barcelona                     Blavis   
3     Barcelona              My Restaurant   
4     Barcelona            Bodega Biarritz   
...         ...                        ...   
1309       Rome                Cambiamenti   
1546       Rome            Pizza & Friends   
2665       Rome  Enoteca Vini Vizi e Virtu   
0        Vienna     Casino Restaurant Wien   
183      Vienna               Senhor Vinho   

                                          cuisine_style  rating  \
0     'International', 'Mediterranean', 'Fusion', 'E...    5.00   
1     'International', 'Mediterranean', 'Spanish', '...    5.00   
2     'Mediterranean', 'European', 'Spanish', 'Veget...    5.00   
3     'Mediterranean', 'European', 'Spanish', 'Veget...    5.00   
4     'Mediterranean', 'Spanish', 'Wine Bar', 'Europ...    5.00   
...                                                 ...     ...   
1309  'Mediterranean', 'Vegetarian Friendly', 'Vegan...    5.00   
1546  'Italian', 'Pizza', 'Mediterranean', 'Vegetari...    5.00   
2665                              'Italian', 'Wine Bar'    5.00   
0     'Austrian', 'European', 'Central European', 'I...    5.00   
183   'Mediterranean', 'European', 'Portuguese', 'Ve...    5.00   

      reviews_number                                            reviews  \
0                792  [['Perfect atmosphere and location', 'Perfecti...   
1               2707  [['Wow! Best ever!', 'Small menu-- GET A RESER...   
2                643  [['The whole package', 'Incredible Service and...   
3                159  [['Fantastic evening!!', 'Perfect meal'], ['01...   
4               1078  [['Deserves its reputation!', 'THESE ARE THE T...   
...              ...                                                ...   
1309             125  [['So perfect ! Elegant and well executed foo....   
1546             123  [['Excellent and creative pizza al taglio', 'N...   
2665             132  [['Not a restaurant, but the loveliest wine s....   
0                241  [["Fine dining at it's best", 'Xmas dinner'], ...   
183              140  [['Portugal in Vienna', 'Excellent authentic P...   

     price_range  
0        Costoso  
1          Medio  
2          Medio  
3          Medio  
4          Medio  
...          ...  
1309     Costoso  
1546      Barato  
2665       Medio  
0        Costoso  
183        Medio  

[119 rows x 7 columns]

In [291]:
cocinas['city2'] = cocinas['city']
cocinas2 = cocinas.set_index('city')
cocinas2

place  \
city                                   
Barcelona                        Uma   
Barcelona                      Viana   
Barcelona                     Blavis   
Barcelona              My Restaurant   
Barcelona            Bodega Biarritz   
...                              ...   
Rome                     Cambiamenti   
Rome                 Pizza & Friends   
Rome       Enoteca Vini Vizi e Virtu   
Vienna        Casino Restaurant Wien   
Vienna                  Senhor Vinho   

                                               cuisine_style  rating  \
city                                                                   
Barcelona  'International', 'Mediterranean', 'Fusion', 'E...    5.00   
Barcelona  'International', 'Mediterranean', 'Spanish', '...    5.00   
Barcelona  'Mediterranean', 'European', 'Spanish', 'Veget...    5.00   
Barcelona  'Mediterranean', 'European', 'Spanish', 'Veget...    5.00   
Barcelona  'Mediterranean', 'Spanish', 'Wine Bar', 'Europ...    5.00   
...                                                      ...     ...   
Rome       'Mediterranean', 'Vegetarian Friendly', 'Vegan...    5.00   
Rome       'Italian', 'Pizza', 'Mediterranean', 'Vegetari...    5.00   
Rome                                   'Italian', 'Wine Bar'    5.00   
Vienna     'Austrian', 'European', 'Central European', 'I...    5.00   
Vienna     'Mediterranean', 'European', 'Portuguese', 'Ve...    5.00   

           reviews_number                                            reviews  \
city                                                                           
Barcelona             792  [['Perfect atmosphere and location', 'Perfecti...   
Barcelona            2707  [['Wow! Best ever!', 'Small menu-- GET A RESER...   
Barcelona             643  [['The whole package', 'Incredible Service and...   
Barcelona             159  [['Fantastic evening!!', 'Perfect meal'], ['01...   
Barcelona            1078  [['Deserves its reputation!', 'THESE ARE THE T...   
...                   ...                                                ...   
Rome                  125  [['So perfect ! Elegant and well executed foo....   
Rome                  123  [['Excellent and creative pizza al taglio', 'N...   
Rome                  132  [['Not a restaurant, but the loveliest wine s....   
Vienna                241  [["Fine dining at it's best", 'Xmas dinner'], ...   
Vienna                140  [['Portugal in Vienna', 'Excellent authentic P...   

          price_range      city2  
city                              
Barcelona     Costoso  Barcelona  
Barcelona       Medio  Barcelona  
Barcelona       Medio  Barcelona  
Barcelona       Medio  Barcelona  
Barcelona       Medio  Barcelona  
...               ...        ...  
Rome          Costoso       Rome  
Rome           Barato       Rome  
Rome            Medio       Rome  
Vienna        Costoso     Vienna  
Vienna          Medio     Vienna  

[119 rows x 7 columns]

In [294]:
#Maximo numero de reviews en esta lista
reviews = cocinas.groupby(['city'])[['city2','price_range','reviews_number']].max()
reviews

city2     price_range  reviews_number
city                                                
Barcelona  Barcelona           Medio            2707
Berlin        Berlin           Medio             423
Lisbon        Lisbon           Medio             423
London        London           Medio            1730
Madrid        Madrid           Medio             352
Milan          Milan           Medio             212
Paris          Paris           Medio            2305
Prague        Prague  SinInformacion             967
Rome            Rome  SinInformacion            4867
Vienna        Vienna           Medio             241

In [314]:
tmp2 = (pd.merge(reviews, cocinas2, on='reviews_number', how='left')).sort_values(by = 'reviews_number',ascending = False)
tmp2 = tmp2.set_index(np.arange(len(tmp2.index)))
tmp2

tmp2.rename(columns={'city2_x': 'Ciudad',
            'place': 'Restaurant Sugerido',
            'price_range_x': 'Rango de precio',
            'cuisine_style': 'Estilo de Cocina',
            'reviews_number': 'Numero de Reviews'}, inplace=True)
tmp2

Ciudad Rango de precio  Numero de Reviews  \
0        Rome  SinInformacion               4867   
1   Barcelona           Medio               2707   
2       Paris           Medio               2305   
3      London           Medio               1730   
4      Prague  SinInformacion                967   
5      Berlin           Medio                423   
6      Berlin           Medio                423   
7      Lisbon           Medio                423   
8      Lisbon           Medio                423   
9      Madrid           Medio                352   
10     Vienna           Medio                241   
11      Milan           Medio                212   

                     Restaurant Sugerido  \
0                          Pane e Salame   
1                                  Viana   
2                                Epicure   
3                      Bar 61 Restaurant   
4               Wine O'Clock Shop Prague   
5                           Naveena Path   
6                 Ha Tapas no Mercado!!!   
7                           Naveena Path   
8                 Ha Tapas no Mercado!!!   
9                                   Cebo   
10                Casino Restaurant Wien   
11  Shaolin State of Harmony - Raw Vegan   

                                     Estilo de Cocina  rating  \
0     'Italian', 'Street Food', 'Vegetarian Friendly'    5.00   
1   'International', 'Mediterranean', 'Spanish', '...    5.00   
2   'French', 'European', 'Vegetarian Friendly', '...    5.00   
3   'Mediterranean', 'European', 'Spanish', 'Veget...    5.00   
4   'Italian', 'Mediterranean', 'European', 'Wine ...    5.00   
5   'Indian', 'Asian', 'Sri Lankan', 'Vegetarian F...    5.00   
6   'European', 'Portuguese', 'Soups', 'Mediterran...    5.00   
7   'Indian', 'Asian', 'Sri Lankan', 'Vegetarian F...    5.00   
8   'European', 'Portuguese', 'Soups', 'Mediterran...    5.00   
9   'Mediterranean', 'European', 'Spanish', 'Conte...    5.00   
10  'Austrian', 'European', 'Central European', 'I...    5.00   
11  'Healthy', 'Soups', 'Vegan Options', 'Vegetari...    5.00   

                                              reviews price_range_y    city2_y  
0   [['Very nice panini', 'The best'], ['01/07/201...        Barato       Rome  
1   [['Wow! Best ever!', 'Small menu-- GET A RESER...         Medio  Barcelona  
2   [['Very nice place', 'Speechless'], ['01/08/20...       Costoso      Paris  
3   [['A Gem of a Tapas Bar!', 'Very good'], ['01/...         Medio     London  
4   [['Best dinner in prague!', 'Lovely evening'],...         Medio     Prague  
5   [['Mindblowing!', 'Authentic Indian food'], ['...        Barato     Berlin  
6   [['Lunch', 'Great meal, fantastic value!'], ['...        Barato     Lisbon  
7   [['Mindblowing!', 'Authentic Indian food'], ['...        Barato     Berlin  
8   [['Lunch', 'Great meal, fantastic value!'], ['...        Barato     Lisbon  
9   [['Absolutely divine tastes and flavours', 'Lo...       Costoso     Madrid  
10  [["Fine dining at it's best", 'Xmas dinner'], ...       Costoso     Vienna  
11  [['THE BEST RAW FOOD EVER', 'A Taste of Harmon...         Medio      Milan

### Dónde es seguro comer:
- Los restaurants a continuación, tienen un promedio de **5** en su rating, todos tienen más comentarios de la media de comentarios de estas ciudades, por lo que serán una **apuesta segura** a la hora de visitarles! 
- Se aprecia en su columna de de reviews un sentimiento positivo por parte de los comensales y todos estos rest se encuentran en un rango medio de precio.

In [315]:
tmp2.loc[:,['Ciudad','Rango de precio','Numero de Reviews','Restaurant Sugerido','Estilo de Cocina','reviews']]

Ciudad Rango de precio  Numero de Reviews  \
0        Rome  SinInformacion               4867   
1   Barcelona           Medio               2707   
2       Paris           Medio               2305   
3      London           Medio               1730   
4      Prague  SinInformacion                967   
5      Berlin           Medio                423   
6      Berlin           Medio                423   
7      Lisbon           Medio                423   
8      Lisbon           Medio                423   
9      Madrid           Medio                352   
10     Vienna           Medio                241   
11      Milan           Medio                212   

                     Restaurant Sugerido  \
0                          Pane e Salame   
1                                  Viana   
2                                Epicure   
3                      Bar 61 Restaurant   
4               Wine O'Clock Shop Prague   
5                           Naveena Path   
6                 Ha Tapas no Mercado!!!   
7                           Naveena Path   
8                 Ha Tapas no Mercado!!!   
9                                   Cebo   
10                Casino Restaurant Wien   
11  Shaolin State of Harmony - Raw Vegan   

                                     Estilo de Cocina  \
0     'Italian', 'Street Food', 'Vegetarian Friendly'   
1   'International', 'Mediterranean', 'Spanish', '...   
2   'French', 'European', 'Vegetarian Friendly', '...   
3   'Mediterranean', 'European', 'Spanish', 'Veget...   
4   'Italian', 'Mediterranean', 'European', 'Wine ...   
5   'Indian', 'Asian', 'Sri Lankan', 'Vegetarian F...   
6   'European', 'Portuguese', 'Soups', 'Mediterran...   
7   'Indian', 'Asian', 'Sri Lankan', 'Vegetarian F...   
8   'European', 'Portuguese', 'Soups', 'Mediterran...   
9   'Mediterranean', 'European', 'Spanish', 'Conte...   
10  'Austrian', 'European', 'Central European', 'I...   
11  'Healthy', 'Soups', 'Vegan Options', 'Vegetari...   

                                              reviews  
0   [['Very nice panini', 'The best'], ['01/07/201...  
1   [['Wow! Best ever!', 'Small menu-- GET A RESER...  
2   [['Very nice place', 'Speechless'], ['01/08/20...  
3   [['A Gem of a Tapas Bar!', 'Very good'], ['01/...  
4   [['Best dinner in prague!', 'Lovely evening'],...  
5   [['Mindblowing!', 'Authentic Indian food'], ['...  
6   [['Lunch', 'Great meal, fantastic value!'], ['...  
7   [['Mindblowing!', 'Authentic Indian food'], ['...  
8   [['Lunch', 'Great meal, fantastic value!'], ['...  
9   [['Absolutely divine tastes and flavours', 'Lo...  
10  [["Fine dining at it's best", 'Xmas dinner'], ...  
11  [['THE BEST RAW FOOD EVER', 'A Taste of Harmon...

In [373]:
citieslist = df_filtered.groupby('city')

madrid = citieslist.get_group('Madrid')
madrid.describe()

rating  reviews_number
count 8656.00         9377.00
mean     3.79           84.80
std      0.75          255.25
min      0.00            0.00
25%      3.50            4.00
50%      4.00           18.00
75%      4.50           70.00
max      5.00        11497.00

In [376]:
madrid.cuisine_style = madrid.cuisine_style.str.replace('[', '')
madrid.cuisine_style = madrid.cuisine_style.str.replace(']', '')
madrid.head(n=10)

name    city  \
0                                     Cebo  Madrid   
1                                   DSTAGE  Madrid   
2                    Los Montes de Galicia  Madrid   
3                                    Astor  Madrid   
4  Restaurante Nuevo Horno de Santa Teresa  Madrid   
5                                  Burnout  Madrid   
6                              Salmon Guru  Madrid   
7                                     Toga  Madrid   
8                                   Maitia  Madrid   
9                             Grama Lounge  Madrid   

                                       cuisine_style ranking  rating  \
0  'Mediterranean', 'European', 'Spanish', 'Conte...    1.00    5.00   
1  'International', 'European', 'Spanish', 'Conte...    2.00    4.50   
2  'International', 'Mediterranean', 'European', ...    3.00    4.50   
3  'International', 'Mediterranean', 'European', ...    4.00    5.00   
4                         'Mediterranean', 'Spanish'    5.00    4.50   
5  'American', 'Bar', 'International', 'Healthy',...    6.00    4.50   
6                       'Bar', 'Vegetarian Friendly'    7.00    4.50   
7  'Mediterranean', 'European', 'Spanish', 'Veget...    8.00    4.50   
8  'Spanish', 'European', 'Vegetarian Friendly', ...    9.00    4.50   
9  'Latin', 'Bar', 'Venezuelan', 'Vegetarian Frie...   10.00    4.50   

  price_range  reviews_number  \
0     Costoso             352   
1     Costoso            1724   
2       Medio            3795   
3       Medio             302   
4       Medio             357   
5       Medio             413   
6       Medio             160   
7       Medio             377   
8     Costoso             690   
9       Medio             337   

                                             reviews  \
0  [['Absolutely divine tastes and flavours', 'Lo...   
1  [['Birthday meal', 'Outstanding experience. Ve...   
2  [['Wonderful lunch', 'One of Madrid’s finest.'...   
3  [['Small place with big flavours', 'The best d...   
4  [['Be hungry, be very hungry....', 'Good good'...   
5  [['When in Spain', 'Best Burger and Service in...   
6  [['Best cocktails evaaah!', 'Great'], ['01/03/...   
7  [['Well balanced', 'Fantastic food and drink c...   
8  [['Top Notch!', 'Excellent experience!'], ['12...   
9  [['Capitán Jack lives here!', 'Great Latin Ame...   

                                              URL_TA      ID_TA  
0  /Restaurant_Review-g187514-d10428302-Reviews-C...  d10428302  
1  /Restaurant_Review-g187514-d6884911-Reviews-DS...   d6884911  
2  /Restaurant_Review-g187514-d2006562-Reviews-Lo...   d2006562  
3  /Restaurant_Review-g187514-d11896546-Reviews-A...  d11896546  
4  /Restaurant_Review-g187514-d1953725-Reviews-Re...   d1953725  
5  /Restaurant_Review-g187514-d11953286-Reviews-B...  d11953286  
6  /Restaurant_Review-g187514-d10731641-Reviews-S...  d10731641  
7  /Restaurant_Review-g187514-d11955452-Reviews-T...  d11955452  
8  /Restaurant_Review-g187514-d3948218-Reviews-Ma...   d3948218  
9  /Restaurant_Review-g187514-d12216862-Reviews-G...  d12216862

In [381]:
(madrid.groupby(['price_range'])[['name']].count()).sort_values(by='name',ascending = False)

name
price_range         
SinInformacion  4498
Medio           3535
Barato          1044
Costoso          300

In [382]:
madrid2 = madrid.groupby(['price_range'])
madridbarato = madrid2.get_group('Barato')
madridbarato = madridbarato.sort_values(by=['rating','reviews_number'],ascending = [False,False])
madridbarato = madridbarato.set_index(np.arange(len(madridbarato.index)))
mbaratos = list(madridbarato['name'][0:15])
madridbarato.head(n=15)


name    city  \
0                                  Smok Mok  Madrid   
1                          La Mula Plateada  Madrid   
2                                    Plenti  Madrid   
3                               Sushiberico  Madrid   
4                                  Keyaan's  Madrid   
5                                      Kint  Madrid   
6                              Madriles Bar  Madrid   
7                 Macanudo Bar Espacio Arte  Madrid   
8                            BeeHealthyShop  Madrid   
9                            Medri Tiramisu  Madrid   
10                         La Creperia Rosa  Madrid   
11  Frutas Prohibidas Juice Smoothie Coffee  Madrid   
12                                  Wok Sun  Madrid   
13                           La Pepa Maluca  Madrid   
14                         Denay Coffee Bar  Madrid   

                                        cuisine_style ranking  rating  \
0   'Asian', 'Street Food', 'Vegetarian Friendly',...  551.00    5.00   
1   'Pizza', 'Argentinean', 'South American', 'Veg...  534.00    5.00   
2   'Cafe', 'Healthy', 'European', 'Vegetarian Fri...  436.00    5.00   
3            'Sushi', 'Spanish', 'Japanese', 'Fusion' 2592.00    5.00   
4   'Caribbean', 'Latin', 'Contemporary', 'Fast Fo...  451.00    5.00   
5        'Vegetarian Friendly', 'Gluten Free Options'  731.00    5.00   
6   'Spanish', 'Vegetarian Friendly', 'Gluten Free...  927.00    5.00   
7   'Brew Pub', 'Fast Food', 'Vegetarian Friendly'...  644.00    5.00   
8   'Asian', 'Taiwanese', 'Fusion', 'Vegetarian Fr... 1965.00    5.00   
9                                           'Italian'  921.00    5.00   
10        'French', 'European', 'Vegetarian Friendly'  713.00    5.00   
11  'Healthy', 'Vegetarian Friendly', 'Gluten Free...  880.00    5.00   
12                                          'Chinese' 1208.00    5.00   
13                                          'Spanish' 3215.00    5.00   
14                          'Bar', 'Pub', 'Gastropub' 1952.00    5.00   

   price_range  reviews_number  \
0       Barato              74   
1       Barato              65   
2       Barato              63   
3       Barato              59   
4       Barato              58   
5       Barato              48   
6       Barato              39   
7       Barato              38   
8       Barato              38   
9       Barato              36   
10      Barato              33   
11      Barato              32   
12      Barato              32   
13      Barato              30   
14      Barato              25   

                                              reviews  \
0   [['Amazing food, amazing people!!', 'premium s...   
1   [['Fantastic empanadas and pizzas, friendly a....   
2   [['aMazInG', 'Excellent eggs for breakfast'], ...   
3   [['Delicious new take on sushi'], ['12/05/2015']]   
4   [['Great taste and service', 'Excellent empana...   
5   [["If you're a coeliac in Madrid, this is the....   
6                                            [[], []]   
7   [['A bar with a nice vibe!', 'Great bar with g...   
8   [['Delicious!', 'I nice break'], ['06/17/2017'...   
9   [['Really good! I mean really really good!', '...   
10              [['Just beautiful!'], ['07/19/2017']]   
11  [['Perfect breakfast', 'Awesome bowles, toasts...   
12  [['Very good', 'Great Asian fast food'], ['06/...   
13  [['Great place to enjoy a cold beer or a glas....   
14                                           [[], []]   

                                               URL_TA      ID_TA  
0   /Restaurant_Review-g187514-d12215222-Reviews-S...  d12215222  
1   /Restaurant_Review-g187514-d8466797-Reviews-La...   d8466797  
2   /Restaurant_Review-g187514-d12400540-Reviews-P...  d12400540  
3   /Restaurant_Review-g187514-d7689202-Reviews-Su...   d7689202  
4   /Restaurant_Review-g187514-d11756470-Reviews-K...  d11756470  
5   /Restaurant_Review-g187514-d11445456-Reviews-K...  d11445456  
6   /Restaurant_Review-g187514-d10496969-Reviews-M...  d104969

In [383]:
madridmedio = madrid2.get_group('Medio')
madridmedio = madridmedio.sort_values(by=['rating','reviews_number'],ascending = [False,False])
madridmedio = madridmedio.set_index(np.arange(len(madridmedio.index)))
mmedio = list(madridmedio['name'][0:15])
madridmedio.head(n=15)

name    city  \
0                            Astor  Madrid   
1                   Meson Molinero  Madrid   
2                       Bizikletak  Madrid   
3      Malvon Empanadas Argentinas  Madrid   
4                    Picnic Madrid  Madrid   
5                  Bombardino Cafe  Madrid   
6   The Original Globet Croquettes  Madrid   
7                          Enklima  Madrid   
8               El Rincon del Cabo  Madrid   
9            Mercado Jamon Iberico  Madrid   
10              La Cocina de Gucho  Madrid   
11             Brew Wild Pizza Bar  Madrid   
12          La Escalera De Serrano  Madrid   
13               Mercado Tragantua  Madrid   
14                       Francisca  Madrid   

                                        cuisine_style ranking  rating  \
0   'International', 'Mediterranean', 'European', ...    4.00    5.00   
1                             'Steakhouse', 'Spanish'   84.00    5.00   
2   'Mediterranean', 'Fusion', 'Vegetarian Friendl...  474.00    5.00   
3   'Argentinean', 'Vegetarian Friendly', 'Vegan O...  172.00    5.00   
4   'Mediterranean', 'European', 'Spanish', 'Veget...  296.00    5.00   
5   'Italian', 'Mediterranean', 'European', 'Spani...  327.00    5.00   
6              'Mediterranean', 'Spanish', 'European' 2313.00    5.00   
7                'Fusion', 'Mediterranean', 'Spanish'  341.00    5.00   
8   'Mediterranean', 'Spanish', 'Gastropub', 'Arge...  137.00    5.00   
9                                           'Spanish'  243.00    5.00   
10  'Mediterranean', 'Spanish', 'European', 'Veget...  352.00    5.00   
11  'Brew Pub', 'Pizza', 'Italian', 'Vegetarian Fr...  311.00    5.00   
12               'Mediterranean', 'Diner', 'Wine Bar'  547.00    5.00   
13  'Mediterranean', 'Spanish', 'Fusion', 'Vegetar...  145.00    5.00   
14                                          'Spanish'  892.00    5.00   

   price_range  reviews_number  \
0        Medio             302   
1        Medio             201   
2        Medio             151   
3        Medio             134   
4        Medio             107   
5        Medio              99   
6        Medio              94   
7        Medio              84   
8        Medio              77   
9        Medio              73   
10       Medio              71   
11       Medio              62   
12       Medio              62   
13       Medio              59   
14       Medio              56   

                                              reviews  \
0   [['Small place with big flavours', 'The best d...   
1   [['Delicious Lamb And Excellent Service', 'Gre...   
2   [['Absolutely wonderful.', 'Hidden gem'], ['05...   
3   [['Great place for a snack', "Best empanadas I...   
4   [['Way beyond my expectations', 'Excellent fro...   
5   [['Great service, food and atmosphere', 'Great...   
6   [['The best croquettes!', 'Delicious'], ['10/0...   
7   [['A wonderful surprise', 'Fantastic food!'], ...   
8   [['Modern tapas', 'Fantastic restaurant/bar'],...   
9   [['Ham to die for', 'Big range of Jamon'], ['1...   
10  [['lovely place and amazing food!', 'Amazon Fo...   
11  [['Yes and yes!', 'Super cozy bar for beer lov...   
12  [['Excellent food. Lovely staff.', 'Great plac...   
13  [['SO GOOD!', 'Delicious food in a local atmos...   
14  [['Basque cooking at its bedt', 'Carmen is inc...   

                                               URL_TA      ID_TA  
0   /Restaurant_Review-g187514-d11896546-Reviews-A...  d11896546  
1   /Restaurant_Review-g187514-d990935-Reviews-Mes...    d990935  
2   /Restaurant_Review-g187514-d10488043-Reviews-B...  d10488043  
3   /Restaurant_Review-g187514-d11511433-Reviews-M...  d11511433  
4   /Restaurant_Review-g187514-d7898096-Reviews-Pi...   d7898096  
5   /Restaurant_Review-g187514-d10355629-Reviews-B...  d10355629  
6   /Restaurant_Review-g187514-d9822092-Reviews-Th...   d9822092  
7   /Restaurant_Review-g187514-d11870922-Reviews-E...  d11870922  
8   /Restaurant_Review-g187514-d12208439-Reviews-E...  d1220843

In [384]:
madridcostos = madrid2.get_group('Costoso')
madridcostos = madridcostos.sort_values(by=['rating','reviews_number'],ascending = [False,False])
madridcostos = madridcostos.set_index(np.arange(len(madridcostos.index)))
mcostoso = list(madridcostos['name'][0:15])
madridcostos.head(n=15)

name    city  \
0                                 Cebo  Madrid   
1   Restaurante Etimo by Begona Fraire  Madrid   
2          Restaurante Crudito by Emil  Madrid   
3                              Arrayan  Madrid   
4                           Cantespino  Madrid   
5          Restaurante Clos Madrid S.L  Madrid   
6                          Kitchenclub  Madrid   
7                  Machete Restaurante  Madrid   
8                           Sinatra Ya  Madrid   
9                       El Santo Baron  Madrid   
10                        Sabor de mar  Madrid   
11                         Gran Clavel  Madrid   
12            The Loft by Salsaisarten  Madrid   
13                 Taller Silk & Spice  Madrid   
14                   Catering Domenico  Madrid   

                                        cuisine_style ranking  rating  \
0   'Mediterranean', 'European', 'Spanish', 'Conte...    1.00    5.00   
1   'Spanish', 'Vegetarian Friendly', 'Gluten Free...  138.00    5.00   
2   'Japanese', 'Spanish', 'Fusion', 'Vegetarian F...  667.00    5.00   
3                          'Mediterranean', 'Spanish' 1024.00    5.00   
4                                           'Spanish' 3527.00    5.00   
5                                           'Spanish' 1309.00    5.00   
6                                      'Contemporary' 2814.00    5.00   
7               'Peruvian', 'Asian', 'South American' 1657.00    5.00   
8                                          'Japanese' 1772.00    5.00   
9                                   'Asian', 'Fusion' 2451.00    5.00   
10                                          'Seafood' 2684.00    5.00   
11                                          'Spanish' 4098.00    5.00   
12                                     'Contemporary' 4105.00    5.00   
13                                   'French', 'Thai' 4143.00    5.00   
14  'International', 'Mediterranean', 'Delicatesse... 4206.00    5.00   

   price_range  reviews_number  \
0      Costoso             352   
1      Costoso              52   
2      Costoso              31   
3      Costoso              24   
4      Costoso              22   
5      Costoso              14   
6      Costoso              13   
7      Costoso              11   
8      Costoso              11   
9      Costoso               6   
10     Costoso               5   
11     Costoso               3   
12     Costoso               0   
13     Costoso               0   
14     Costoso               0   

                                              reviews  \
0   [['Absolutely divine tastes and flavours', 'Lo...   
1   [['A meal to remember, excellent experience!',...   
2   [['Beautiful uniquely Spanish Japanese fusion'...   
3   [['Amazing dinner', 'Excellent choice!!!'], ['...   
4                                            [[], []]   
5                                            [[], []]   
6   [['Amazing unforgettable experience'], ['05/19...   
7   [['Excellent', 'Machete!!!'], ['11/28/2017', '...   
8   [['Great Food, Great Japanese Wagyu Beef grea....   
9                                            [[], []]   
10                                           [[], []]   
11                  [['Great tapas'], ['12/31/2017']]   
12                                           [[], []]   
13                                           [[], []]   
14                                           [[], []]   

                                               URL_TA      ID_TA  
0   /Restaurant_Review-g187514-d10428302-Reviews-C...  d10428302  
1   /Restaurant_Review-g187514-d13138415-Reviews-R...  d13138415  
2   /Restaurant_Review-g187514-d12097619-Reviews-R...  d12097619  
3   /Restaurant_Review-g187514-d12608132-Reviews-A...  d12608132  
4   /Restaurant_Review-g187514-d1991075-Reviews-Ca...   d1991075  
5   /Restaurant_Review-g187514-d13297733-Reviews-R...  d13297733  
6   /Restaurant_Review-g187514-d3598523-Reviews-Ki...   d3598523  
7   /Restaurant_Review-g187514-d13174390-Reviews-M...  d13174390  
8  

In [385]:
madridsininfo = madrid2.get_group('SinInformacion')
madridsininfo = madridsininfo.sort_values(by=['rating','reviews_number'],ascending = [False,False])
madridsininfo = madridsininfo.set_index(np.arange(len(madridsininfo.index)))
msininfo = list(madridsininfo['name'][0:15])
madridsininfo.head(n=15)

name    city       cuisine_style ranking  rating  \
0        Taberna la Ilustracion  Madrid   'Cafe', 'Spanish'  521.00    5.00   
1                   Tabernicola  Madrid                 NaN  851.00    5.00   
2                        Astral  Madrid                 NaN  939.00    5.00   
3   Meson Restaurante El Molino  Madrid  'Spanish', 'Grill'  632.00    5.00   
4                 Chulapas Cafe  Madrid        'Bar', 'Pub' 1829.00    5.00   
5              La Buena Cerveza  Madrid                 NaN 2117.00    5.00   
6         La Cigala Restaurante  Madrid                 NaN 2153.00    5.00   
7           Dreams Cocktail Bar  Madrid                 NaN 1991.00    5.00   
8                    Estar Cafe  Madrid                 NaN 1902.00    5.00   
9                     Meat Love  Madrid                 NaN 1147.00    5.00   
10           Bar Sidreria Ruval  Madrid                 NaN 2013.00    5.00   
11                   La Violeta  Madrid                 NaN 1745.00    5.00   
12                      Lemme's  Madrid                 NaN 2649.00    5.00   
13           La Buena Mesa Peru  Madrid                 NaN 1356.00    5.00   
14       Las Tapas de Cervantes  Madrid                 NaN 1860.00    5.00   

       price_range  reviews_number  \
0   SinInformacion              69   
1   SinInformacion              48   
2   SinInformacion              41   
3   SinInformacion              37   
4   SinInformacion              31   
5   SinInformacion              30   
6   SinInformacion              28   
7   SinInformacion              27   
8   SinInformacion              26   
9   SinInformacion              25   
10  SinInformacion              23   
11  SinInformacion              20   
12  SinInformacion              19   
13  SinInformacion              18   
14  SinInformacion              17   

                                              reviews  \
0                                            [[], []]   
1   [['Fabulous foods and fantastic hospitality f....   
2                                            [[], []]   
3                                            [[], []]   
4                                            [[], []]   
5   [['Great selection of beer', 'Cool Place'], ['...   
6                                            [[], []]   
7   [['Ridiculously amazing', 'Cocktails from heav...   
8                                            [[], []]   
9   [['Good Service', 'Nice Burger'], ['12/22/2017...   
10        [['Friendly and charming'], ['09/09/2016']]   
11                                           [[], []]   
12                                           [[], []]   
13  [['Authentic Peruvian Food in Madrid'], ['04/2...   
14  [['Great visit', 'Excellet!!!'], ['12/07/2017'...   

                                               URL_TA      ID_TA  
0   /Restaurant_Review-g187514-d7284083-Reviews-Ta...   d7284083  
1   /Restaurant_Review-g187514-d10063815-Reviews-T...  d10063815  
2   /Restaurant_Review-g187514-d991697-Reviews-Ast...    d991697  
3   /Restaurant_Review-g187514-d12367309-Reviews-M...  d12367309  
4   /Restaurant_Review-g187514-d4366743-Reviews-Ch...   d4366743  
5   /Restaurant_Review-g187514-d7696133-Reviews-La...   d7696133  
6   /Restaurant_Review-g187514-d12221524-Reviews-L...  d12221524  
7   /Restaurant_Review-g187514-d4503939-Reviews-Dr...   d4503939  
8   /Restaurant_Review-g187514-d7252091-Reviews-Es...   d7252091  
9   /Restaurant_Review-g187514-d12922250-Reviews-M...  d12922250  
10  /Restaurant_Review-g187514-d5795001-Reviews-Ba...   d5795001  
11  /Restaurant_Review-g187514-d10336215-Reviews-L...  d10336215  
12  /Restaurant_Review-g187514-d8362902-Reviews-Le...   d8362902  
13  /Restaurant_Review-g187514-d12112688-Reviews-L...  d12112688  
14  /Restaurant_Review-g187514-d10635288-Reviews-L...  d10635288

In [386]:
print('Sugerencias para comer en madrid segun los rangos de precio: ')
print('Costosos: ',mcostoso,'\nMedios: ',mmedio,'\nBaratos: ',mbaratos,'\nSinInfo: ',msininfo)


Sugerencias para comer en madrid segun los rangos de precio: 
Costosos:  ['Cebo', 'Restaurante Etimo by Begona Fraire', 'Restaurante Crudito by Emil', 'Arrayan', 'Cantespino', 'Restaurante Clos Madrid S.L', 'Kitchenclub', 'Machete Restaurante', 'Sinatra Ya', 'El Santo Baron', 'Sabor de mar', 'Gran Clavel', 'The Loft by Salsaisarten', 'Taller Silk & Spice', 'Catering Domenico'] 
Medios:  ['Astor', 'Meson Molinero', 'Bizikletak', 'Malvon Empanadas Argentinas', 'Picnic Madrid', 'Bombardino Cafe', 'The Original Globet Croquettes', 'Enklima', 'El Rincon del Cabo', 'Mercado Jamon Iberico', 'La Cocina de Gucho', 'Brew Wild Pizza Bar', 'La Escalera De Serrano', 'Mercado Tragantua', 'Francisca'] 
Baratos:  ['Smok Mok', 'La Mula Plateada', 'Plenti', 'Sushiberico', "Keyaan's", 'Kint', 'Madriles Bar', 'Macanudo Bar Espacio Arte', 'BeeHealthyShop', 'Medri Tiramisu', 'La Creperia Rosa', 'Frutas Prohibidas Juice Smoothie Coffee', 'Wok Sun', 'La Pepa Maluca', 'Denay Coffee Bar'] 
SinInfo:  ['Taberna l

In [423]:
tmp5 = madridcostos.append(madridmedio, ignore_index=True).append(madridbarato,ignore_index = True).append(madridsininfo,ignore_index = True)
tmp5 = tmp5.set_index(np.arange(len(tmp5.index)))
tmp5 = tmp5.dropna(subset=['cuisine_style']) 
tmp5.reset_index(drop=True,inplace=True) 
tmp5

name    city  \
0                                   Cebo  Madrid   
1     Restaurante Etimo by Begona Fraire  Madrid   
2            Restaurante Crudito by Emil  Madrid   
3                                Arrayan  Madrid   
4                             Cantespino  Madrid   
...                                  ...     ...   
6283                          El Secreto  Madrid   
6284                        El Del Medio  Madrid   
6285                            Fit Food  Madrid   
6286                         La Revuelta  Madrid   
6287                         Casa Carmen  Madrid   

                                          cuisine_style ranking  rating  \
0     'Mediterranean', 'European', 'Spanish', 'Conte...    1.00    5.00   
1     'Spanish', 'Vegetarian Friendly', 'Gluten Free...  138.00    5.00   
2     'Japanese', 'Spanish', 'Fusion', 'Vegetarian F...  667.00    5.00   
3                            'Mediterranean', 'Spanish' 1024.00    5.00   
4                                             'Spanish' 3527.00    5.00   
...                                                 ...     ...     ...   
6283                                 'Spanish', 'Diner'     nan     nan   
6284                'Spanish', 'Contemporary', 'Fusion'     nan     nan   
6285                                          'Spanish'     nan     nan   
6286                                          'Spanish'     nan     nan   
6287                         'Mediterranean', 'Spanish'     nan     nan   

         price_range  reviews_number  \
0            Costoso             352   
1            Costoso              52   
2            Costoso              31   
3            Costoso              24   
4            Costoso              22   
...              ...             ...   
6283  SinInformacion               0   
6284  SinInformacion               0   
6285  SinInformacion               0   
6286  SinInformacion               0   
6287  SinInformacion               0   

                                                reviews  \
0     [['Absolutely divine tastes and flavours', 'Lo...   
1     [['A meal to remember, excellent experience!',...   
2     [['Beautiful uniquely Spanish Japanese fusion'...   
3     [['Amazing dinner', 'Excellent choice!!!'], ['...   
4                                              [[], []]   
...                                                 ...   
6283                                                NaN   
6284                                                NaN   
6285                                                NaN   
6286                                                NaN   
6287                                                NaN   

                                                 URL_TA      ID_TA  
0     /Restaurant_Review-g187514-d10428302-Reviews-C...  d10428302  
1     /Restaurant_Review-g187514-d13138415-Reviews-R...  d13138415  
2     /Restaurant_Review-g187514-d12097619-Reviews-R...  d12097619  
3     /Restaurant_Review-g187514-d12608132-Reviews-A...  d12608132  
4     /Restaurant_Review-g187514-d1991075-Reviews-Ca...   d1991075  
...                                                 ...        ...  
6283  /Restaurant_Review-g187514-d13328766-Reviews-E...  d13328766  
6284  /Restaurant_Review-g187514-d13329973-Reviews-E...  d13329973  
6285  /Restaurant_Review-g187514-d13331427-Reviews-F...  d13331427  
6286  /Restaurant_Review-g187514-d13343539-Reviews-L...  d13343539  
6287  /Restaurant_Review-g187514-d13345511-Reviews-C...  d13345511  

[6288 rows x 10 columns]